Change to accept both date and startDate / endDate combination

chashuramen opened the pull request:
Fixes #1

chashuramen commented:
Changes:

112-147, 203-246 in `sqs/receiver.ts`

26-29, 39-41 in  `sqs/messages/AddMessage.ts `

abid-roomboss commented:
Niceeeeeeeee, @chashuramen possible to add a test with mocking both sides.
Only if its easy.

chashuramen commented:
I was wondering this as I didn't see any test specs in sepc/ . How do I add one?

abid-roomboss commented:
There isn't anything yet, so freedom to explore any option.

chashuramen commented:
Wrapped in #3.

chashuramen commented:
Switched From tslint (now deprecated) to eslint-typescript for linting.
Added prettier for enhanced code formatting.
Added Jest for testing. Babel to assist code transformation on tests.

To Lint: `npm run-script lint` (does not work, currently has too many errors).
To Test: `npm test`.

chashuramen commented:
Should be safe to remove the following (but didn't remove):

dependencies:
```
"@types/jasmine": "^3.6.3",
"jasmine": "^3.6.4",
"jasmine-spec-reporter": "^6.0.0",
"jasmine-ts": "^0.3.0",
"nyc": "^15.1.0",
"tslint": "^5.20.0",
"tslint-config-airbnb": "^5.11.2",
```

files:
```
sqs/helpers/isoDate.js
spec/
jasmine.json
tslint.json
```

chashuramen commented:
Turned out the test setup is more complicated than I thought.

JavaScript sometimes makes me cryヾ(｡>﹏<｡)ﾉ !

abid-roomboss commented:
Hi @chashuramen 

Thanks for this.

Should test be passing right now! I tried in local but failed however report says passed, .
![image](https://user-images.githubusercontent.com/47208386/110262923-77980300-7ff8-11eb-97de-b31285e00659.png)

abid-roomboss commented:
Ah may be thats coverage report?
